google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineEnvironment.java

```diff
@@ -23,7 +23,8 @@
  */
 public enum AppEngineEnvironment {
   APP_ENGINE_STANDARD("appengine.environment.name.standard"),
-  APP_ENGINE_FLEX("appengine.environment.name.flexible");
+  APP_ENGINE_FLEX("appengine.environment.name.flexible"),
+  APP_ENGINE_FLEX_COMPAT("appengine.environment.name.flexible");
```

Comment:
Yeah, I know what you mean. I just fail to see an occasion where a discussion in such terms would take place.

The other point I was making is that by renaming AppEngineEnvironment to something else like AppEngineProjectType also makes it harder to understand environment <-> project type comparisons, for example. The semantic loss there would be bigger and more frequent in my opinion.

Yeah, I meant rolling back to whatever existed before. The rationale for adding this new FLEX_COMPAT"environment"/"project type" was to get rid of checks like "if targetEnv.isFlexible or isFlexCompat", which feel a bit hacky. But I'm wondering if we can really get rid of them, I think the dynamic nature of the flexCompat checks makes them necessary.